.github/workflows/check-roles.yml

```diff
@@ -0,0 +1,61 @@
+name: Check Roles
+
+on:
+  schedule:
+    - cron: '30 8,16 * * 1,2,3,4,5'
+  workflow_dispatch:
+
+jobs:
+  check-roles:
+    runs-on: moj-cloud-platform
+    strategy:
+      matrix:
+        environment: [test, preprod, prod]
+        project: [pre-sentence-reports-to-delius, prison-case-notes-to-probation, workforce-allocations-to-delius]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Map GitHub environment to values file
+        id: values
+        shell: bash
+        run: |
+          if [ '${{ matrix.environment }}' == 'test' ];    then echo '::set-output name=filename::values-dev.yml'; fi
+          if [ '${{ matrix.environment }}' == 'preprod' ]; then echo '::set-output name=filename::values-preprod.yml'; fi
+          if [ '${{ matrix.environment }}' == 'prod' ];    then echo '::set-output name=filename::values-prod.yml'; fi
+
+      - name: Get Hmpps Auth vals
+        id: hmpps-auth-values
+        shell: bash
+        run: |
+          project_name=$(echo '${{ matrix.project }}' | tr '[:lower:]' '[:upper:]' | sed 's/-/_/g')
+          echo "::set-output name=token_url::$(cat ${{ steps.values.outputs.filename }} | yq '.env.SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_HMPPS-AUTH_TOKEN-URI')
+          echo "::set-output name=roles::$(cat values.yml | yq '.required-roles')
+          echo "::set-output name=client_id_name::${project_name}_CLIENT_ID"
+          echo "::set-output name=client_secret_name::${project_name}_CLIENT_SECRET"
+        working-directory: projects/${{ matrix.project }}/deploy
+
+      - name: Get Hmmps Auth Roles
+        id: hmpps-auth-request
+        env:
+          CLIENT_ID_NAME: ${{ steps.hmpps-auth-values.outputs.client_id_name }}
+          CLIENT_SECRET_NAME: ${{ steps.hmpps-auth-values.outputs.client_secret_name }}
+        uses: fjogeleit/http-request-action@v1
```

Comment:
We should probably use a specific sha for this - otherwise we're potentially exposing our secrets to `fjogeleit` if they retag their repo with some malicious code.